apache/flink-docker
# [FLINK-15793] Support native K8s mode

This PR will integrate flink`kubernetes-entry.sh` into the unified flink-docker `docker-entrypoint.sh`. After then users can enable build-in plugins much more easily. 

Using the following command to enable the plugin.
```
flink run-application -t kubernetes-application -Dkubernetes.cluster-id=k8s-ha-app1 ...
-Dcontainerized.master.env.ENABLE_BUILT_IN_PLUGINS=flink-oss-fs-hadoop-1.11.1.jar \
-Dcontainerized.taskmanager.env.ENABLE_BUILT_IN_PLUGINS=flink-oss-fs-hadoop-1.11.1.jar \
local:///opt/flink/examples/streaming/StateMachineExample.jar
```

# How to verify this PR?
Update Flink code
* Use `docker-entrypoint.sh` as entrypoint of JobManager and TaskManager pod
* Add `native-k8s` as an argument

Then run a Flink native K8s session or application, it should deploy successfully just as before.
